verify.go

```diff
@@ -90,8 +90,13 @@ func verifySignatureAtTime(p7 *PKCS7, signer signerInfo, truststore *x509.CertPo
 			return err
 		}
 		h := hash.New()
-		h.Write(p7.Content)
-		computed := h.Sum(nil)
+		var computed []byte
+		if p7.HashCalc != nil {
+			computed = p7.HashCalc(h, p7.Content)
```

Comment:
This would then become `computed = p7.Hasher.Hash(h, p7.Content)`.

Can you define an unexported function on `p7` (e.g. `p7.calculateHash`) that performs the hashing, so that you only have the `if/else` defined once?